ddtrace/propagation/http.py

```diff
@@ -1140,9 +1126,8 @@ def my_controller(url, headers):
         :param dict headers: HTTP headers to extract tracing attributes.
         :return: New `Context` with propagated attributes.
         """
-        context = Context()
-        if not headers:
-            return context
+        if not headers or not config._propagation_style_extract:
+            return Context()
```

Comment:
Should we still initialize a context object at the top of the function?

```suggestion
        context = Context()
        if not headers or not config._propagation_style_extract:
            return context
```

Reply:
Def